merlin/systems/dag/ops/implicit.py

```diff
@@ -0,0 +1,119 @@
+#
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import importlib
+import json
+import pathlib
+
+from merlin.dag import ColumnSelector  # noqa
+from merlin.schema import ColumnSchema, Schema
+from merlin.systems.dag.ops.operator import InferenceDataFrame, PipelineableInferenceOperator
+
+
+class PredictImplicit(PipelineableInferenceOperator):
+    """Operator for running inference on Implicit models.."""
+
+    def __init__(self, model, **kwargs):
```

Comment:
The `PredictTensorflow` op has `model_or_path` as its param and will load the model from a path if that's what is provided. I think we should try to be consistent with the constructor functionality, and `PredictTensorflow` is probably the most mature/popular, so matching that seems like a good idea to me.

Reply:
👍 I agree that having consistency across these Predict ops is a good idea. 

However, the implicit serialization format doesn't appear to have a mechanism for knowing which model to load. 

e.g. if you have created a model and saved to a file, for example `model.npz`.

```python
import implicit

# random training data
n = 100  # number of both users and items
user_items = csr_matrix(np.random.choice([0, 1], size=n * n, p=[0.9, 0.1]).reshape(n, n))

model = implicit.bpr.BayesianPersonalizedRanking()
model.fit(user_items)
model.save("model.npz")  # save model to file
```

to call the `load` method we require knowledge of the kind of model to be loaded. There's not an `implicit.load` function that I can see (correct me if there is something I've missed @benfred )

```python
implicit.gpu.bpr.BayesianPersonalizedRanking.load("model.npz")
```

We could change the signature to `model_or_path` at a later stage if the functionality becomes available and that is our preferred option for these predict ops.